shimmingtoolbox/load_nifti.py

```diff
@@ -0,0 +1,165 @@
+                break
+            else:
+                logging.error("Input must be linked to an acquisition folder. {} is out of range".format(input_resp))
+
+        nifti_path = os.path.abspath(file_list[select_acquisition])
+
+    nifti_list = [os.path.join(nifti_path, f) for f in os.listdir(nifti_path) if f.endswith((".nii", ".nii.gz"))]
+    n_echos = len(nifti_list)
+
+    if n_echos <= 0:
+        raise RuntimeError("No acquisition images in selected path {}".format(nifti_path))
+
+    info_init, json_init, img_init = read_nii(nifti_list[0])
+
+    info = []
+    json_info = []
+    niftis = np.empty([1, 1], dtype=float)
+    print(info_init.shape)
+    if info_init.ndim == 3:
+        niftis = np.empty([info_init.shape[0], info_init.shape[1], info_init.shape[2], n_echos, 1], dtype=float)
+        for i_echo in range(n_echos):
+            tmp_nii = read_nii(os.path.abspath(nifti_list[i_echo]))
+            info.append(tmp_nii[0].header)
+            json_info.append(tmp_nii[1])
+            niftis[:, :, :, i_echo, 0] = tmp_nii[2]
+    else:
+        niftis = np.empty([
+            info_init.shape[1], info_init.shape[2], info_init.shape[3],
+            n_echos, info_init.shape[0]], dtype=float)
+        for i_echo in range(n_echos):
+            tmp_nii = read_nii(os.path.abspath(nifti_list[i_echo]))
+            info.append(tmp_nii[0].header)
+            json_info.append(tmp_nii[1])
+            for i_volume in range(info_init.shape[0]):
+                niftis[:, :, :, i_echo, i_volume] = tmp_nii[2][i_volume, :, :, :]
+
+    return niftis, info, json_info
+
+
+def read_nii(nii_path, auto_scale=True):
+    """ Reads a nifti file and returns the corresponding image and info. Also returns the associated json data.
+    Args:
+        nii_path (str): direct path to the .nii or .nii.gz file that is going to be read
+        auto_scale (:obj:`bool`, optional): Tells if scaling is done before return
+    Returns:
+        info (Nifti1Image): Objet containing various data about the nifti file (returned by nibabel.load)
+        json_data (dict): Contains the different fields present in the json file corresponding to the nifti file
+        image (ndarray): Image contained in the read nifti file. Siemens phase images are rescaled between 0 and 2pi.
+    """
+
+    info = nib.load(nii_path)
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    json_path = nii_path.split('.nii')[0] + '.json'
+
+    if os.path.isfile(json_path):
+        json_data = json.load(open(json_path))
+    else:
+        raise ValueError('Missing json file')
+
+    image = np.asarray(info.dataobj)
+    if auto_scale:
+        if ('Manufacturer' in json_data) and (json_data['Manufacturer'] == 'Siemens') \
+                and (image_type(json_data) == 'phase'):
+            image = image * (2 * math.pi / PHASE_SCALING_SIEMENS)
+
+    return info, json_data, image
+
+
+def image_type(json_data):
```

Comment:
Can nibabel output this for us instead?

Reply:
I didn't see a image type field in the nibabel header and nibabel cannot open json type files. So the information has to be loaded with another way than nibabel for what I've seen. Maybe I've just missed the field to do it.

Reply:
@po09i what do we want from this `image_type()` function? what would be example of output "type"?

Reply:
It only tells us if it's phase or magnitude from the 'ImageType' section in the json file thta is linked to the nifti file. The read_nii function uses it to know if the data needs to be scaled in radian or not. I did not find a good way in the header of the nibabel file to tell us this information.

Reply:
don't we already have that information after [BIDS conversion](https://github.com/shimming-toolbox/shimming-toolbox-py/blob/master/config/dcm2bids.json#L105)?

Reply:
Yes, but dcm2bids automatically creates the json file that contain this information with the data. So I was trying to make this function independent form dcm2bids so when you only have the original nifti files and json without dcm2bids, you could also know this information.

Reply:
ah i see. I think at this point in the development i would suggest to not bother covering all possible scenario. It is fair to assume input data is either DICOM or BIDS folder, in which case no need for the `image_type()` function.